osu.Game/Beatmaps/ControlPoints/ControlPointInfo.cs

```diff
@@ -230,32 +250,12 @@ public static T BinarySearch<T>(IReadOnlyList<T> list, double time)
         {
             ArgumentNullException.ThrowIfNull(list);
 
-            if (list.Count == 0)
-                return null;
-
-            if (time < list[0].Time)
-                return null;
-
-            if (time >= list[^1].Time)
-                return list[^1];
+            int index = BinarySearchUtils.BinarySearch(list, time, c => c.Time, EqualitySelection.Rightmost);
```

Comment:
Just for a bit of commentary, I think this better matches osu!stable ([`ControlPointAt`](https://github.com/peppy/osu-stable-reference/blob/bb57924c1552adbed11ee3d96cdcde47cf96f2b6/osu!/GameplayElements/Beatmaps/Beatmap.cs#L1477-L1482) / [`BeatLengthAt`](https://github.com/peppy/osu-stable-reference/blob/bb57924c1552adbed11ee3d96cdcde47cf96f2b6/osu!/GameplayElements/Beatmaps/Beatmap.cs#L1392-L1410))

Will run a diffcalc to be sure.